python/examples/atomistic/running-ase-md.py

```diff
@@ -0,0 +1,269 @@
+#
+# .. math::
+#
+#     E_i = k * (\vec{r_i} - \vec{r_i}^0)^2,
+#
+# where :math:`k` is a harmonic force constant, :math:`\vec{r_i}` the position of atom
+# :math:`i` and :math:`\vec{r_i}^0` the position of atom :math:`i` at equilibrium. The
+# energy of the whole system is then
+#
+# .. math::
+#
+#     E = \sum_i E_i.
+#
+# .. _Einstein's solid: https://en.wikipedia.org/wiki/Einstein_solid
+#
+# Let's implement this model! You can check the :ref:`atomistic-tutorial-export`
+# tutorial for more information on how to define your own custom models.
+
+
+class HarmonicModel(torch.nn.Module):
+    def __init__(self, k: float, equilibrium_positions: torch.Tensor):
+        """Create an ``HarmonicModel``.
+
+        :param k: force constant, in ``energy unit / (length unit)^2``
+        :param equilibrium_positions: torch tensor with shape ``n x 3``, containing the
+            equilibrium positions of all atoms
+        """
+        super().__init__()
+        assert k > 0
+        self.k = k
+        self.equilibrium_positions = equilibrium_positions
+
+    def forward(
+        self,
+        systems: List[System],
+        outputs: Dict[str, ModelOutput],
+        selected_atoms: Optional[Labels],
+    ) -> Dict[str, TensorMap]:
+        # if the model user did not request an energy calculation, we have nothing to do
+        if "energy" not in outputs:
+            return {}
+
+        # we don't want to worry about selected_atoms yet
+        if selected_atoms is not None:
+            raise NotImplementedError("selected_atoms is not implemented")
+
+        if outputs["energy"].per_atom:
+            raise NotImplementedError("per atom energy is not implemented")
+
+        # compute the energy for each system by adding together the energy for each atom
+        energy = torch.zeros((len(systems), 1), dtype=systems[0].positions.dtype)
+        for i, system in enumerate(systems):
+            assert len(system) == self.equilibrium_positions.shape[0]
+            energy[i] += torch.sum(
+                self.k * (system.positions - self.equilibrium_positions) ** 2
+            )
+
+        # add metadata to the output
+        systems_id = torch.arange(len(systems), dtype=torch.int32)
+        block = TensorBlock(
+            values=energy,
+            samples=Labels("system", systems_id.reshape(-1, 1)),
+            components=[],
+            properties=Labels("energy", torch.IntTensor([[0]])),
+        )
+        return {
+            "energy": TensorMap(
+                keys=Labels("_", torch.IntTensor([[0]])), blocks=[block]
+            )
+        }
+
+
+# %%
+#
+# Initial simulation state
+# ------------------------
+#
+# Now that we have a model for the energy of our system, let's create some initial
+# simulation state. We'll build a 3x3x3 super cell of diamond carbon. In pactise, you
+# could also read the initial state from a file.
+
+primitive = ase.build.bulk("C", "diamond", a=3.567)
```

Comment:
maybe explicitly give the parameter names here

```suggestion
primitive = ase.build.bulk(name="C", crystalstructure="diamond", a=3.567)
```